magento/pwa-studio
# PWA-209: Track Page Views

## Description

Added events in order to track page views.

**Acceptance Criteria**

- Page View event is tracked when a user lands on a page upon entering the site and when a user navigates to a new page.
- Page View event contains pagetype:
  - page
  - PLP
  - PDP
  - cart
  - checkout
  - order confirmation
- For each pageType collect specific to this page type attributes
- Example extension subscribes to the events and has events available for processing
- Example extension simply subscribes to event and logs its occurrence

## Related Issue







Closes #[PWA-2097](https://jira.corp.magento.com/browse/PWA-2097).

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps






1. Load Venia homepage
2. Open browser console and verify that the sample extension logged the event (with its contextual data)
3. Navigate to a different page
4. Open browser console and verify that the sample extension logged the event (with its contextual data)
5. Go offline
6. Navigate back to the homepage
7. Open browser console and verify that the sample extension logged the event (with its contextual data)

#### Test scenario(s) for direct fix/feature





#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
